src/App.js

```diff
@@ -46,6 +49,15 @@ function App() {
           >
             <Route path='/log' element={<DeliveryLogDash />} />
           </Route>
+
+          <Route element={<AuthenticateUser redirectPath={'/posts'} requiredRole={['TEACHER']}/>}>
+            <Route path='/cohorts/add-cohort' element={<AddCohortForm />} />
+          </Route>
+
+          <Route element={<AuthenticateUser redirectPath={'/posts'} requiredRole={['TEACHER']}/>}>
```

Comment:
We don't need to have two of these protected routes. I believe we should be able to add the new `add-cohort` route and the `/cohorts/:id` nested inside one `AuthenticateUser` route.